packages/data-context/src/actions/ProjectActions.ts

```diff
@@ -345,7 +354,24 @@ export class ProjectActions {
     this.api.insertProjectPreferencesToCache(this.ctx.lifecycleManager.projectTitle, args)
   }
 
-  async codeGenSpec (codeGenCandidate: string, codeGenType: CodeGenType): Promise<NexusGenUnions['GeneratedSpecResult']> {
+  async getReactComponentsFromFile (filePath: string): Promise<ReactComponentDescriptor[]> {
+    try {
+      const src = await fs.readFile(filePath, 'utf8')
```

Comment:
```suggestion
      const src = await this.ctx.fs.readFile(filePath, 'utf8')
```

I see a convention where do use `ctx.fs` to access this. WDYT?